From 87972f78d794252aed4f0e102d09b3495a2bc966 Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:56 +0800
Subject: [PATCH] MLK-9617-19 usb: otg: test: show warning message for
 unsupported otg test dev

According to "USB On-The-Go and Embedded Host Automated Compliance Plan Version
1.2", chapter 6.7.23, unknown Device Not Supporting HNP:'Unsupported Device';
chapter 6.7.24 unknown Device Supporting HNP: 'Device No Response'

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 32df700386fff1d1eef90007604055f1d013aaf3)
---
 drivers/usb/core/otg_whitelist.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/usb/core/otg_whitelist.h b/drivers/usb/core/otg_whitelist.h
index b261550..d02811e 100644
--- a/drivers/usb/core/otg_whitelist.h
+++ b/drivers/usb/core/otg_whitelist.h
@@ -84,6 +84,19 @@ static int is_targeted(struct usb_device *dev)
 	     le16_to_cpu(dev->descriptor.idProduct) == 0xbadd))
 		return 0;
 
+	/* Unknown Device Not Supporting HNP */
+	if ((le16_to_cpu(dev->descriptor.idVendor) == 0x1a0a &&
+		le16_to_cpu(dev->descriptor.idProduct) == 0x0201)) {
+		dev_warn(&dev->dev, "Unsupported Device\n");
+		return 0;
+	}
+	/* Unknown Device Supporting HNP */
+	if ((le16_to_cpu(dev->descriptor.idVendor) == 0x1a0a &&
+		le16_to_cpu(dev->descriptor.idProduct) == 0x0202)) {
+		dev_warn(&dev->dev, "Device no Responding\n");
+		return 0;
+	}
+
 	/* NOTE: can't use usb_match_id() since interface caches
 	 * aren't set up yet. this is cut/paste from that code.
 	 */
-- 
1.8.0

